lib/mpool/src/mclass.c

```diff
@@ -207,7 +207,7 @@ mclass_filecnt_get(const char *path, const struct stat *sb, int typeflag, struct
 }
 
 static void
-mclass_destroy_setup(uint8_t filecnt, struct workqueue_struct *wq)
+mclass_destroy_setup(int filecnt, struct workqueue_struct *wq)
```

Comment:
I guess the context I am missing is that somewhere are along the way filecnt changes from a uint8_t (from cparams) to an int now. Why is this the case?

Reply:
A media class directory can contain few other files other than the mblock data files like MDC, mblock metadata file etc. So, when one creates a KVDB with the experimental param storage.<mclass>.file.count set to UINT8_MAX, the total file count in the media class directory exceeds the max value and results in truncation.

Reply:
I feel like it makes more sense to make the cparam a uint16_t with a max of UINT8_MAX. Then mpool can be consistent about a uint16_t everywhere instead of a dynamically sized int. I'll let others comment though.

Definitely should add a functional test regarding the filecnt regardless of solution however. Don't want Sujini's test to get lost to the hands of time.

Reply:
At the very least, the sign of the int should probably be consistent.

Reply:
The truncation issue here has nothing to do with the type of storage.file.count and I won't change that..

This is just an internal function which accepts the number of files in a media class directory as a parameter.

Reply:
Added unit test to cover this scenario, thnx!